discovery.md

```diff
@@ -101,18 +101,295 @@ be considered an event subscriber and a target user of this API.
 
 ## API Specification
 
-This document structure focuses on discovery where each document has a two part
-key of `provider` and `type`. This takes a denormalized view where the composite
-key of documents is the `provider` and `type`. Provider is defined as an
-arbitrary human readable string and `type` is the CloudEvent `type` attribute.
+This API is specified as a REST API with well defined entities and relationships
+between those entities.
 
-### Provider entity Attributes
+All responses will have a content-type of `application/json` and are either
+a single JSON encoded object or an array of objects.
+
+### Entities in the API
+
+1. `Provider`
+2. `Type`
+3. `Producer`
+
+The `Provider` and `Type` entities form the basis of the directory and can be
+used to build user experiences around discovery. The `Producer` entity is
+keyed off of `{Provider.Name, Type.Name}` and provides the necessary details to
+create a subscription for events of that `ce-type` from the selected `provider`.
```

Comment:
s/provider/Provider/

Reply:
Is `matching` an optional feature or are all servers required to support it? As written it's required, IMO. If you meant for it to be optional then we'll need to state that someplace.  I'm leaning towards it being required. We just need to discuss whether it should be that any part of the string contains the search-string, or just prefix or postfix - since some people asked about this on a previous call.